test/init.js

```diff
@@ -4,7 +4,8 @@
 // License text available at https://opensource.org/licenses/Artistic-2.0
 
 'use strict';
-var DataSource = require('loopback-datasource-juggler').DataSource;
+var juggler = require('loopback-datasource-juggler');
```

Comment:
Does this get the root level juggler dev dependency? Would `npm` dedupe cause any issues?

Reply:
Yes, this is getting root-level juggler dev dependency. `npm` dedupe is not causing any issues as far as I was able to tell. Here is the `node_modules` layout:

```
deps
  - juggler-v3
    - node_modules
      - async (v2.6.2)
      - loopback-datasource-juggler (v3.30.0)
  - juggler-v4
    - node_modules
      - async (v2.6.2)
      - debug (v4.1.1)
      - depd (v2.0.0)
      - loopback-datasource-juggler (v4.6.3)
node_modules
  - loopback-datasource-juggler (v4.6.3)
  - juggler-v3 -> ../deps/juggler-v3
  - juggler-v4 -> ../deps/juggler-v4
  - (plus a bunch of other packages)
```

It also means that we have three kinds of tests now:

 - The test suite from juggler@3 using juggler@3 runtime
 - The test suite from juggler@4 using juggler@4 runtime
 - The connector-specific test suite in `test/*.js` always using juggler@4 runtime from top-level `node_modules`.

Reply:
After I removed `node_modules` folder and reinstalled dependencies, I got slightly different layout which is closer to what I would expect to see.

```
deps
  - juggler-v3
    - node_modules
      - async (v2.6.2)
      - loopback-datasource-juggler (v3.30.0)
  - juggler-v4 
    # no node_modules created!
node_modules
  - loopback-datasource-juggler (v4.6.3)
  - juggler-v3 -> ../deps/juggler-v3
  - juggler-v4 -> ../deps/juggler-v4
  - (plus a bunch of other packages)

Reply:
Thanks for checking and clarifying @bajtos, that looks good.